openshift/dynamic/client.py

```diff
@@ -79,25 +79,25 @@ def get_resources_for_apiversion(self, prefix, group, version, preferred):
             )
         return resources
 
-    def list(self, resource, namespace=None):
+    def list(self, resource, namespace=None, label_selector=None, field_selector=None):
         path_params = {}
         if resource.namespaced and namespace:
             resource_path = resource.urls['namespaced_base']
             path_params['namespace'] = namespace
         else:
             resource_path = resource.urls['base']
-        return ResourceInstance(resource, self.request('get', resource_path, path_params=path_params))
+        return ResourceInstance(resource, self.request('get', resource_path, path_params=path_params, label_selector=label_selector, field_selector=field_selector))
 
-    def get(self, resource, name=None, namespace=None):
+    def get(self, resource, name=None, namespace=None, label_selector=None, field_selector=None):
         if name is None:
-            return self.list(resource, namespace=namespace)
+            return self.list(resource, namespace=namespace, label_selector=label_selector, field_selector=field_selector)
```

Comment:
This is a nice thing for you to do but I think it may be easy to be consistent if you force them to provide the resource. Example:

```
$ kubectl get
You must specify the type of resource to get. Valid resource types include:

  * all
  * certificatesigningrequests (aka 'csr')
  * ...
```

Reply:
Yeah, it still requires the resource type, name there refers to the name of the instance. So the resource would be deployment, name could be 'hello-deploy'. If name is not provided it will list deployment configs

Reply:
Yeah, it still requires the resource type, name there refers to the name of the instance. So the resource would be deployment, name could be 'hello-deploy'. If name is not provided it will list deployment configs